javascript/lang/security/audit/code-string-concat.yaml

```diff
@@ -1,34 +1,76 @@
+                - pattern-inside: function ... ($REQ, $RES, $NEXT) {...}
+                - patterns:
+                    - pattern-either:
+                        - pattern-inside: $APP.$METHOD(..., function $FUNC($REQ, $RES) {...})
+                        - pattern-inside: $APP.$METHOD(..., function $FUNC($REQ, $RES, $NEXT) {...})
+                    - metavariable-regex:
+                        metavariable: $METHOD
+                        regex: ^(get|post|put|head|delete|options)$
+            - pattern-either:
+                - pattern: $REQ.query
+                - pattern: $REQ.body
+                - pattern: $REQ.params
+                - pattern: $REQ.cookies
+                - pattern: $REQ.headers
+        - patterns:
+            - pattern-either:
+                - pattern-inside: |
+                    import { ...,$IMPORT,... } from 'next/router'
+                    ...
+                - pattern-inside: |
+                    import $IMPORT from 'next/router';
+                    ...
+            - pattern-either:
+                - patterns:
+                    - pattern-inside: |
+                        $ROUTER = $IMPORT()
+                        ...
+                    - pattern-either:
+                        - pattern-inside: |
+                            const { ...,$PROPS,... } = $ROUTER.query
+                            ...
+                        - pattern-inside: |
+                            var { ...,$PROPS,... } = $ROUTER.query
+                            ...
+                        - pattern-inside: |
+                            let { ...,$PROPS,... } = $ROUTER.query
+                            ...
+                    - focus-metavariable: $PROPS
+                - patterns:
+                    - pattern-inside: |
+                        $ROUTER = $IMPORT()
+                        ...
+                    - pattern: |
+                        $ROUTER.query.$VALUE 
+                - patterns:
+                    - pattern: $IMPORT().query.$VALUE
+    pattern-sinks:
       - patterns:
-          - pattern-either:
-              - patterns:
-                  - pattern: $STRING + $EXPR
-                  - pattern-not: $STRING + "..."
-              - patterns:
-                  - pattern: $EXPR + $STRING
-                  - pattern-not: '"..." + $STRING'
-              - patterns:
-                  - pattern: "[..., $STRING, ...].join(...)"
-              - patterns:
-                  - pattern: $VAR += $STRING
-          - metavariable-regex:
-              metavariable: $STRING
-              regex: .*(function|Function|eval|\([^\(\)]*\)\s*=>).*
-      - patterns:
-          - pattern-inside: |
-              `...${...}...`
-          - pattern-regex: function|Function|eval|\([^\(\)]*\)\s*=>
+        - pattern: |
+            $M(...)
+        - metavariable-regex:
+            metavariable: $M
+            regex: (eval|exec)
```

Comment:
child_process.exec? https://nodejs.org/api/child_process.html probably just keep to eval and save the others for a new rule.. idk

Reply:
removed. can always add a new rule for child_process later